src/index.js

```diff
@@ -1 +1,57 @@
-import './style.css';
\ No newline at end of file
+// index.js
+
+import './style.css';
+
+const apiUrl = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/HITirhqNpaDAqFyG4jlT/scores/';
+
+const fetchData = async () => {
+  const response = await fetch(apiUrl);
+  const data = await response.json();
+  return data.result || [];
+};
+
+const fetchScore = async (user, score) => {
+  await fetch(apiUrl, {
+    method: 'POST',
+    body: JSON.stringify({ user, score }),
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
+};
+
+const displayLists = async () => {
+  try {
+    const lists = await fetchData();
+    const scoreList = document.querySelector('.score-list');
+    scoreList.innerHTML = ''; // Clear existing content before appending new scores
+    lists.forEach(({ user, score }) => {
+      const div = document.createElement('div');
+      div.classList.add('score-list-item'); // Correct class name here
+      div.innerHTML = `<p>${user}: ${score}</p>`;
+      scoreList.appendChild(div);
+    });
+  } catch (error) {
+    console.error('Error fetching data:', error);
+  }
+};
+
+const submitBtn = document.querySelector('.btn');
+const nameForm = document.querySelector('#name');
+const scoreForm = document.querySelector('#score');
+
+submitBtn.addEventListener('click', async (e) => {
+  e.preventDefault();
+  const error = document.querySelector('.error');
+  if (!nameForm.value || !scoreForm.value) {
+    error.innerHTML = 'Please fill all input fields';
+  } else {
+    await fetchScore(nameForm.value, scoreForm.value);
+    nameForm.value = '';
+    scoreForm.value = '';
+    error.innerHTML = '';
+    displayLists();
+  }
+});
+
```

Comment:
- [ ] Right now the refresh button is somewhat useless since the scores are added immediately to the page. When I add a score it updates immediately on the UI. But that is not the requirement. The requirement is that the scores should not be fetched and updated until we click the Refresh button. Only when we click the refresh button should the scores be fetched and displayed. Kindly fix this issue.